iothub_client/src/iothub_client_core_ll.c

```diff
@@ -2319,6 +2336,24 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_LL_SetOption(IOTHUB_CLIENT_CORE_LL_HANDLE
                 result = IOTHUB_CLIENT_OK;
             }
         }
+        else if (strcmp(optionName, OPTION_DT_MODEL_ID) == 0)
+        {
+            if (handleData->dt_model_id != NULL)
```

Comment:
>if (handleData->dt_model_id != NULL) [](start = 12, length = 36)

We shouldn't let callers pass this multiple times, since the hub itself requires tearing down connection on a new DCM.  I think we should fail here instead if already set.